03_extendable_form/README.md

```diff
@@ -0,0 +1,189 @@
+yarn pack 
+shasum -b -a 512 custom-form-example-plugin-v0.3.3.tgz | awk '{ print $1 }' | xxd -r -p | base64 | awk '{ printf("%s", $0) }' | awk '{print "sha512-"$0}' ; 
+```
+
+### Deployment
+Once the dynamic plugin is exposed (either NPM registry or an HTTP server), we can add reference to it into RHDH `dynamic-plugins-rhdh` ConfigMap:
+
+```yaml
+data:
+  dynamic-plugins.yaml: |
+    includes:
+      - dynamic-plugins.default.yaml
+    plugins:
+      - disabled: false
+        package: "https://github.com/rhdhorchestrator/custom-form-example-plugin/releases/download/v0.3.3/custom-form-example-plugin-v0.3.3.tgz"
+        integrity: sha512-hCX6XwWpakV5lO83/zexKdZeiOHXvNLuHaOtoA2PLiwQYuGc9gz75+NEHhaEZ80QUoD1CddoYJjRLqr1rteRLw==
+        pluginConfig:
+          dynamicPlugins:
+            frontend:
+              custom-form-example-plugin:
+                countriesUrl: https://restcountries.com/v3.1/all
+```
+
+And wait till the the operator restarts the pod.
+
+Watch the `backstage-backstage-xxxx` pod logs for potential errors, same as the browser console.
+The init container of this pod is responsible for downloading and installing the dynamic plugin packages, errors are reported in its logs.
+
+## Accessing external services
+Per description above, the `OrchestratorFormApi` is solemnly frontend code, meaning it is executed in the browser.
+
+Using common backstage APIs, the widgets can retrieve data from the Backstage services or call the Backstage API.
+An example of receiving two of the Backstage APIs is in the [example plugin](./custom-form-example-plugin/plugins/custom-form-example-plugin/src/plugin.ts).
+
+Access to 3rd party external services (like the OpenShift API) might be affected by either network or CORS limitations.
+To address these issues, a backstage backend plugin exposing new REST endpoints can be implemented to act as a proxy or pre-processor for big data sets.
+This is out of scope for this demo, we refer to the Backstage documentation for additional details.
+
+## Limitations
+- Frontend - huge data sets, CORS or network limitations
+  - to access 3rd party APIs or pre-processing large datasets, yet another backend plugin exposing new Backstage REST API endpoints should be considered
+
+- Requires TypeScript/React coding skills. Can not be handled declaratively.
+
+- Single `OrchestratorFormApi` implementation per RHDH deployment
+  - widgets for all workflows are supplied via a single plugin, development coordination and development standards are needed
+
+## More info
+- [Orchestrator Extensible Form](https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/orchestrator/docs/extensibleForm.md)
+- [Backstage plugin development](https://backstage.io/docs/plugins/structure-of-a-plugin)
+- [RHDH dynamic plugins development](https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.4/#Extend)
+- [JSON Schema](https://json-schema.org/understanding-json-schema/reference)
```

Comment:
should you include also 
- [react-jsonschema-form](https://github.com/rjsf-team/react-jsonschema-form)